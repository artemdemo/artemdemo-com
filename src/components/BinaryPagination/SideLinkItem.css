.side-link-arrow {
  flex-grow: 1;
}
  .side-link-arrow:first-child {
    padding-right: 5px;
  }
  .side-link-arrow:last-child {
    padding-left: 5px;
  }

.side-link-text {
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}
