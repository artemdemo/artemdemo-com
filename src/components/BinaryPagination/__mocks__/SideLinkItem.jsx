import React from 'react';

export const SIDE_ITEM_DIRECTION = {
    'PREV': 'prev',
    'NEXT': 'next',
};

const SideLinkItem = props => (
    <div data-mock='SideLinkItem'>
        {JSON.stringify(props, null, 2)}
    </div>
);

export default SideLinkItem;