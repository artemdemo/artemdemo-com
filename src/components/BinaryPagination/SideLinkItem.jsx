import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'gatsby';
import './SideLinkItem.css';

export const SIDE_ITEM_DIRECTION = {
  'PREV': 'prev',
  'NEXT': 'next',
};

const SideLinkItem = (props) => {
  const {direction, title, to} = props;
  const renderArrow = () => (
    <span className='side-link-arrow'>
            {direction === SIDE_ITEM_DIRECTION.PREV ? '←' : '→'}
        </span>
  );
  const renderTitle = () => (
    <span className='side-link-text'>
      {title}
    </span>
  );
  return (
    <ul className='pagination binary-pagination'>
      <li className='page-item'>
        <Link
          className='page-link binary-pagination-link'
          to={to}
          rel={direction}
        >
          {
            direction === SIDE_ITEM_DIRECTION.PREV ?
              <>
                {renderArrow()}
                {renderTitle()}
              </> :
              <>
                {renderTitle()}
                {renderArrow()}
              </>
          }
        </Link>
      </li>
    </ul>
  );
};

SideLinkItem.propTypes = {
  direction: PropTypes.oneOf([
    SIDE_ITEM_DIRECTION.PREV,
    SIDE_ITEM_DIRECTION.NEXT,
  ]).isRequired,
  to: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default SideLinkItem;
