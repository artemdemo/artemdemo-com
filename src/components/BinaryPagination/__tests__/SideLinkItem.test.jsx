import React from 'react';
import renderer from 'react-test-renderer';
import SideLinkItem, { SIDE_ITEM_DIRECTION } from '../SideLinkItem';

jest.mock('gatsby');

describe('SideLinkItem', () => {
    it('should render "next"', () => {
        const tree = renderer.create(
            <SideLinkItem
                direction={SIDE_ITEM_DIRECTION.NEXT}
                to='path/to/next'
                title='Next title'
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render "prev"', () => {
        const tree = renderer.create(
            <SideLinkItem
                direction={SIDE_ITEM_DIRECTION.PREV}
                to='path/to/prev'
                title='Prev title'
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
