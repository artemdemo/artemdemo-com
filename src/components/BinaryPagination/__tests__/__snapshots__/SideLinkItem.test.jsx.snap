// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideLinkItem should render "next" 1`] = `
<ul
  className="pagination binary-pagination"
>
  <li
    className="page-item"
  >
    <div
      data-mock="Link"
    >
      {
  "className": "page-link binary-pagination-link",
  "to": "path/to/next",
  "rel": "next"
}
      <span
        className="SideLinkItem__LinkText-sc-1sb2mn5-0 itQixI"
      >
        Next title
      </span>
      <span
        className="SideLinkItem__LinkArrow-sc-1sb2mn5-1 bARCRt"
      >
        →
      </span>
    </div>
  </li>
</ul>
`;

exports[`SideLinkItem should render "prev" 1`] = `
<ul
  className="pagination binary-pagination"
>
  <li
    className="page-item"
  >
    <div
      data-mock="Link"
    >
      {
  "className": "page-link binary-pagination-link",
  "to": "path/to/prev",
  "rel": "prev"
}
      <span
        className="SideLinkItem__LinkArrow-sc-1sb2mn5-1 bARCRt"
      >
        ←
      </span>
      <span
        className="SideLinkItem__LinkText-sc-1sb2mn5-0 itQixI"
      >
        Prev title
      </span>
    </div>
  </li>
</ul>
`;
